managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -1705,7 +1710,8 @@ public void asyncClose(final AsyncCallbacks.CloseCallback callback, final Object
 
         lock.readLock().lock();
         try {
-            if (!individualDeletedMessages.isEmpty()) {
+            if (cursorLedger != null && protobufFormat == ZNodeProtobufFormat.Text
```

Comment:
yes, I understood that on `TextFormat` we only write in ledger as we can't write in znode. However, I meant: if we are writing in `Binary` format and on cursor-close if znode-update fails then we won't have latest `individualDeletedMessages` list in ledger, which is fine and we were not handling it before as well.